packages/eas-cli/src/credentials/ios/actions/new/SetupBuildCredentialsFromCredentialsJson.ts

```diff
@@ -0,0 +1,186 @@
+      appleProvisioningProfile: provisioningProfile,
+    });
+  }
+
+  async getProvisioningProfileToAssignAsync(
+    ctx: Context,
+    targetCredentials: IosTargetCredentials,
+    appleTeam: AppleTeamFragment,
+    currentProvisioningProfile: AppleProvisioningProfileFragment | null
+  ): Promise<AppleProvisioningProfileFragment> {
+    const { provisioningProfile } = targetCredentials;
+
+    if (!currentProvisioningProfile) {
+      return await this.createNewProvisioningProfileAsync(ctx, targetCredentials, appleTeam);
+    }
+
+    const isSameProfile = currentProvisioningProfile.provisioningProfile === provisioningProfile;
+    if (!isSameProfile) {
+      return await this.createNewProvisioningProfileAsync(ctx, targetCredentials, appleTeam);
+    }
+
+    // If the configured Provisioning Profile is the same as the new one, just use the currently configured profile
+    return currentProvisioningProfile;
+  }
+
+  async runAsync(ctx: Context): Promise<IosAppBuildCredentialsFragment> {
+    let localCredentials;
+    try {
+      localCredentials = await readIosCredentialsAsync(ctx.projectDir);
+    } catch (error) {
+      Log.error(
+        'Reading credentials from credentials.json failed. Make sure this file is correct and all credentials are present there.'
+      );
+      throw error;
+    }
+
+    // TODO: implement storing multi-target credentials on EAS servers
+    if (isCredentialsMap(localCredentials)) {
+      throw new Error(
+        'Storing multi-target iOS credentials from credentials.json on EAS servers is not yet supported.'
+      );
+    }
+
+    // currently configured credentials
+    const buildCredentials = await getBuildCredentialsAsync(ctx, this.app, this.distributionType);
+    const currentDistributionCertificate = await getDistributionCertificateAsync(
+      ctx,
+      this.app,
+      this.distributionType
+    );
+    const currentProfile = await getProvisioningProfileAsync(ctx, this.app, this.distributionType);
+    const appInfo = `@${this.app.account.name}/${this.app.projectName} (${this.app.bundleIdentifier})`;
+
+    // new credentials from local json
+    const appleTeamFromProvisioningProfile = readAppleTeam(localCredentials.provisioningProfile);
+    const appleTeam = await ctx.newIos.createOrGetExistingAppleTeamAsync(this.app, {
+      appleTeamIdentifier: appleTeamFromProvisioningProfile.teamId,
+      appleTeamName: appleTeamFromProvisioningProfile.teamName,
+    });
+    if (buildCredentials) {
+      Log.log('Currently configured credentials:');
+      displayProjectCredentials(this.app, buildCredentials);
+      if (!ctx.nonInteractive) {
+        const confirm = await confirmAsync({
+          message: `Would you like to replace this configuration with credentials from credentials json?`,
+        });
+        if (!confirm) {
+          throw new Error('Aborting setup of build credentials from credentials json');
+        }
+      }
```

Comment:
Maybe we should actually compare whether the credentials in the .json are different than the ones on our servers? If they are the same then there's nothing to replace.

Reply:
I just saw the message below ("Credentials have already been configured to your credentials json."). However, I think this check should happen before asking the question as if they're the same the question seems irrelevant.

Reply:
yeah that's a pretty good idea, great catch !